bot/exts/info/doc/_cog.py

```diff
@@ -0,0 +1,467 @@
+        await self.bot.wait_until_guild_available()
+        await self.refresh_inventory()
+
+    async def update_single(
+        self, api_package_name: str, base_url: str, inventory_url: str
+    ) -> bool:
+        """
+        Rebuild the inventory for a single package.
+
+        Where:
+            * `package_name` is the package name to use, appears in the log
+            * `base_url` is the root documentation URL for the specified package, used to build
+                absolute paths that link to specific symbols
+            * `inventory_url` is the absolute URL to the intersphinx inventory.
+
+        If the inventory file is currently unreachable,
+        the update is rescheduled to execute in 2 minutes on the first attempt, and 5 minutes on subsequent attempts.
+
+        Return True on success; False if fetching failed and was rescheduled.
+        """
+        self.base_urls[api_package_name] = base_url
+        package = await fetch_inventory(self.bot.http_session, inventory_url)
+
+        if not package:
+            delay = 2*60 if inventory_url not in self.scheduled_inventories else 5*60
+            log.info(f"Failed to fetch inventory, attempting again in {delay//60} minutes.")
+            self.inventory_scheduler.schedule_later(
+                delay,
+                api_package_name,
+                fetch_inventory(self.bot.http_session, inventory_url)
+            )
+            self.scheduled_inventories.add(api_package_name)
+            return False
+        with suppress(KeyError):
+            self.scheduled_inventories.discard(api_package_name)
+
+        for group, items in package.items():
+            for symbol, relative_doc_url in items:
+                if "/" in symbol:
+                    continue  # skip unreachable symbols with slashes
+
+                group_name = group.split(":")[1]
+                if (original_symbol := self.doc_symbols.get(symbol)) is not None:
+                    if group_name in FORCE_PREFIX_GROUPS:
+                        symbol = f"{group_name}.{symbol}"
+                        self.renamed_symbols.add(symbol)
+
+                    elif (original_symbol_group := original_symbol.group) in FORCE_PREFIX_GROUPS:
+                        overridden_symbol = f"{original_symbol_group}.{symbol}"
+                        if overridden_symbol in self.renamed_symbols:
+                            overridden_symbol = f"{api_package_name}.{overridden_symbol}"
+
+                        self.doc_symbols[overridden_symbol] = original_symbol
+                        self.renamed_symbols.add(overridden_symbol)
+
+                    elif api_package_name in PRIORITY_PACKAGES:
+                        self.doc_symbols[f"{original_symbol.package}.{symbol}"] = original_symbol
+                        self.renamed_symbols.add(symbol)
+
+                    else:
+                        symbol = f"{api_package_name}.{symbol}"
+                        self.renamed_symbols.add(symbol)
```

Comment:
There could (and probably are) some edge cases where we get shared symbols inside packages that are outside of the FORCE_PREFIX_GROUPS, causing this to chain the package name (previously they'd just be overwritten), is taking care of that worth the complexity?

Reply:
What's your idea for how to fix it?